src/components/UserList.js

```diff
@@ -1,23 +1,67 @@
 import React from 'react';
+import {connect} from 'react-redux'
 import UserItem from './UserItem';
+import { addUser } from '../actions';
 
-class UserList extends React.Component {
-    render() {
-        return (
-            <>
-                <form>
-                    <div>
-                        <input />
-                        <input type="submit" value="dodaj" />
-                    </div>
-                </form>
-
-                <ul>
-                    <UserItem />
-                </ul>
-            </>
-        )
+const mapStateToProps = (state, props) => {
+    return {
+        users: state.users,
     }
 }
+const mapActionToProps = { addUser };
+
+class UserList extends React.Component {
+	state = {
+		user: '',
+	};
+
+	handleOnChange = (e) => {
+		this.setState({ user: e.target.value });
+	};
+
+	handleAddUser = (e) => {
+		e.preventDefault();
+		const id = this.setID();
```

Comment:
raczej nazwa funkcji to `generateID` lub `createID` ostatecznie `getNextId` - set mi tu nie pasuje bo zazwyczaj używamy tej nazwy do ustawiania ID :P